EmaPajic/assignment2/assignment2_Q1.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+@author: EmaPajic
+"""
+import unittest
+
+"""If we assume that we can have a pointer to the parent"""
+
+class TreeNode:
+    def __init__(self, val, left = None, right = None, parent = None):
+        self.val = val
+        self.left = left
+        self.right = left
+        self.parent = parent
+        
+"""We just go through pointers to parent until we reach None"""
+def find_ancestors(node):
+    
+    result = []
+    if node == None:
+        return None
+    temp = node
+    while temp.parent != None:
+        result.append(temp.parent.val)
+        temp = temp.parent
+    return result
+
+"""Function to build tree from preorder that contains None pointers where needed"""
```

Comment:
Can you factor out the common functions and classes in another file to avoid duplicating code?

Reply:
updated :)